### PR TITLE
move to jsr330

### PR DESCRIPTION
**Pull request by ebarboni**
Replace the component annotation by jsr330 injection.
MavenProject and MavenSession have Parameters equivalent.

### CODE REVIEW

File: nb-repository-plugin/src/main/java/org/apache/netbeans/nbm/repository/PopulateRepositoryMojo.java [hunk 1]
```diff
@@ -247,19 +247,23 @@ public class PopulateRepositoryMojo
...
      *
      */
-    @Component
     private ArtifactHandlerManager artifactHandlerManager;
 
     // parent handler in case we have one
     private Parent artefactParent = null;
 
+    @Inject
+    public PopulateRepositoryMojo(ArtifactHandlerManager artifactHandlerManager,RepositorySystem repositorySystem) {
```
Comment (mbien) [inline nb-repository-plugin/src/main/java/org/apache/netbeans/nbm/repository/PopulateRepositoryMojo.java:262]: space after `,`?

File: nbm-maven-plugin/src/main/java/org/apache/netbeans/nbm/CreateClusterAppMojo.java [hunk 2]
```diff
@@ -176,15 +176,19 @@ public class CreateClusterAppMojo
...
-    @Component
-    private ArtifactResolver artifactResolver;
+    private final ArtifactResolver artifactResolver;
 
-    @Component
+    @Parameter(defaultValue = "${session}", required = true, readonly = true)
     protected MavenSession session;
 
+    @Inject
+    public CreateClusterAppMojo(ArtifactFactory artifactFactory,ArtifactResolver artifactResolver) {
```
Comment (mbien) [inline nbm-maven-plugin/src/main/java/org/apache/netbeans/nbm/CreateClusterAppMojo.java:187]: space after `,`?

File: nbm-maven-plugin/src/main/java/org/apache/netbeans/nbm/CreateNbmMojo.java [hunk 3]
```diff
@@ -178,19 +178,25 @@ public class CreateNbmMojo
...
+    private final MavenProjectHelper projectHelper;
 
-    @Component
-    private Map<String, ArtifactRepositoryLayout> layouts;
+    private final Map<String, ArtifactRepositoryLayout> layouts;
 
     private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("yyyy/MM/dd");
 
+    @Inject
+    public CreateNbmMojo(ArtifactFactory artifactFactory, MavenProjectHelper projectHelper, Map<String, ArtifactRepositoryLayout> layouts,MavenResourcesFiltering mavenResourcesFiltering) {
```
Comment (mbien) [inline nbm-maven-plugin/src/main/java/org/apache/netbeans/nbm/CreateNbmMojo.java:192]: space after `,`?

**Review by mbien — APPROVED** [review]
awesome! looks good to me.

**Comment by ebarboni** [comment]
lets merge then

**Comment by mbien** [comment]
i think it would be good to squash PRs like this one in future before merge to have the history reduced

**Comment by ebarboni** [comment]
sorry for that

**Pull request merged**